### ISSUE TITLE
Compiler unit tests are never run in CI

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#148266]
I tried this code on a checkout branched from f5e2df741b4a9820a7579f0c8eccc951706a8782:

```sh
$ x test compiler/rustc_parse_format
```

I expected to see this happen: No warnings.

Instead, this happened:
```
Testing stage1 {rustc_parse_format} (aarch64-apple-darwin)
   Compiling rustc_index v0.0.0 (/Users/jyn/src/rust-upstream/compiler/rustc_index)
warning: extern crate `smallvec` is unused in crate `rustc_index`
 --> compiler/rustc_index/src/lib.rs:2:1
  |
2 | #![cfg_attr(all(feature = "nightly", test), feature(stmt_expr_attributes))]
  | ^
  |
  = help: remove the dependency or add `use smallvec as _;` to the crate root
  = note: requested on the command line with `-W unused-crate-dependencies`

```

This in itself isn't a big deal, but it seems bad that we aren't running tests that would have caught this ... I spot-checked the logs for the latest bors merge (https://github.com/rust-lang/rust/actions/runs/18907890296/job/53970647517) and couldn't find `rustc_parse_format` anywhere in them.

**Comment by ehuss** [comment]
The only way to fully test this would be to use something like `cargo hack` (or have bootstrap do that), which I think would be very expensive.

An alternative would be to use cargo's unstable [feature unification](https://doc.rust-lang.org/cargo/reference/unstable.html#feature-unification) to force all features to be used, but I would not advise that.

> latest bors merge (https://github.com/rust-lang/rust/actions/runs/18907890296/job/53970647517) and couldn't find `rustc_parse_format` anywhere in them.

It should be in every test job. The output looks like:

```
     Running unittests src/lib.rs (obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/rustc_parse_format-0d34affbc57d2d93)
```
